web/src/components/tasks/NaicsCodeInput.tsx

```diff
@@ -101,15 +104,19 @@ export const NaicsCodeInput = (props: Props): ReactElement => {
   };
 
   const handleChange = (value: string): void => {
-    setNaicsCode(value);
-    if (value.length === LENGTH) {
-      if (getCode(value)) {
-        setIsInvalid(undefined);
+    if (isAuthenticated === IsAuthenticated.FALSE && business?.profileData.industryId === "generic") {
+      setShowNeedsAccountModal(true);
+    } else {
```

Comment:
[pebble] I'd prefer to follow our our patterns and put the isAuthenticated if-check into a guard clause instead of putting the main body of the function into an else-clause.